ui/components/app/currency-input/hooks/useStateWithFirstTouch.tsx

```diff
@@ -0,0 +1,16 @@
+import { useState } from 'react';
+
+// symbol to cover edge case where initial value is literally 'INITIAL_VALUE'
+const INITIAL_VALUE = Symbol('INITIAL_VALUE');
+
+export default function useStateWithFirstTouch<T>(
```

Comment:
Which bug does this address?

Reply:
It lets us know if the token amount has been modified by a user input in `CurrencyInput` so we can know if the amount is being updated upstream after opening the send flow via the "EDIT" flow